contrib/guix/README.md

```diff
@@ -9,79 +11,835 @@ downloads.
 
 We achieve bootstrappability by using Guix as a functional package manager.
 
-## Requirements
+# Requirements
 
 Conservatively, a x86_64 machine with:
 
 - 16GB of free disk space on the partition that /gnu/store will reside in
 - 8GB of free disk space per platform triple you're planning on building (see
   the `HOSTS` environment variable description)
 
-## Setup
+# Installation and Setup
+
+At the time of writing, the latest Guix release is v1.2.0, which has a known
+problems w/re building GnuTLS from source. Solutions and workarounds are
+documented [here](), but I would recommend that you install v1.2.1 (planned for
+release on April 18th, 2021) or later if your choice of installation option
+allows.
+
+This only needs to be done once per machine. If you have already completed the
+installation and setup, please proceed to [perform a build](#usage).
+
+Otherwise, you may choose from one of the following options to install Guix:
+
+1. Using the official **shell installer script** [⤓ skip to section]()
+   - Maintained by Guix developers
+   - Easiest (automatically performs *most* setup)
+   - Works on nearly all Linux distributions
+   - Only installs latest release
+   - Binary installation only, requires high level of trust
+   - Note: The script needs to be run as root, so it should be inspected before it's run
+2. Using the official **binary tarball** [⤓ skip to section]()
+   - Maintained by Guix developers
+   - Normal difficulty (full manual setup required)
+   - Works on nearly all Linux distributions
+   - Installs any release
+   - Binary installation only, requires high level of trust
+3. Using fanquake's **Docker image** [↗︎ external instructions]()
+   - Maintained by fanquake
+   - Easy (automatically performs *some* setup)
+   - Works on most Linux distributions
+   - Installs any release
+   - Binary installation only, requires high level of trust
+4. Using a **distribution-maintained package** [⤓ skip to section]()
+   - Maintained by distribution's Guix package maintainer
+   - Normal difficulty (manual setup required)
+   - Works only on distributions with Guix packaged, at time of writing:
```

Comment:
I don't think we should list the systems that have a guix package available. This will quickly diverge from reality. For example, a quick search shows openSuse should be added to the list: https://software.opensuse.org/package/guix

Reply:
I do think that we can give facts "at time of writing," sometimes it's better than nothing. However, in this particular case, perhaps we can simply link to https://repology.org/project/guix/versions ?